src/main/java/com/tratif/pdfgen/document/builders/PageBuilder.java

```diff
@@ -0,0 +1,62 @@
+package com.tratif.pdfgen.document.builders;
+
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.templateresolver.StringTemplateResolver;
+
+import java.io.StringWriter;
+import java.util.Map;
+
+public class PageBuilder {
+
+    private static TemplateEngine TEMPLATE_ENGINE;
```

Comment:
hide thymeleaf classes behind an internal HtmlRenderer interface. It can simplify unit testing, but even if it's not needed, it will decouple us from thymeleaf -- will be easier to extend with other template renderers (in the future)